pages/mintKamon/index.tsx

```diff
@@ -31,10 +31,10 @@ const MintKamon: NextPage = () => {
   const [tokenURI, setTokenURI] = useState('')
 
   useEffect(() => {
-    setTokenURI('httsp://yourtokenURI') // TODO: for enable and mintWithHenkaku method
+    // setTokenURI('httsp://yourtokenURI') // TODO: for enable and mintWithHenkaku method
   }, [])
 
-  if (false && !tokenURI) {
+  if (mounted && data?.address && !tokenURI) {
     return <GenerateImageForm />
```

Comment:
to setTokenURI, in easy way, would be pass GenerateImageForm setTokenURI like this way

<GenerateImageForm setTokenURI={setTokenURI} />

in GenerateImageForm use setTokenURI

wdyt ? @hourglasshoro

Reply:
I really appreciate any tips on this part! It sounds like passing information from the component to the page shouldn't be too complex, but I have never done that on next or react so any tips might really save me a lot of research time.

Reply:
Yes, I think we often do pass callback functions. For example, onGenerateImage or something.

I think it would be better to create a handleGenerateImage function on the page side and set URI within the handleGenerateImage function. Also, if you know the behavior of useCallback, you can use it for better performance.